components/datetime/src/time_zone.rs

```diff
@@ -147,26 +162,31 @@ impl TimeZoneFormat {
                 _ => None,
             });
 
-        let mut exemplar_cities: Option<DataPayload<provider::time_zones::ExemplarCitiesV1Marker>> =
-            None;
-        let mut mz_generic_long: Option<
-            DataPayload<provider::time_zones::MetaZoneGenericNamesLongV1Marker>,
-        > = None;
-        let mut mz_generic_short: Option<
-            DataPayload<provider::time_zones::MetaZoneGenericNamesShortV1Marker>,
-        > = None;
-        let mut mz_specific_long: Option<
-            DataPayload<provider::time_zones::MetaZoneSpecificNamesLongV1Marker>,
-        > = None;
-        let mut mz_specific_short: Option<
-            DataPayload<provider::time_zones::MetaZoneSpecificNamesShortV1Marker>,
-        > = None;
-
+        let mut tz_format: TimeZoneFormat = Self {
+            kind: TimeZoneFormatKind::Pattern(patterns.clone()),
+            data_payloads,
+            locale,
+            format_units,
+            fallback_unit: TimeZoneFormat::get_fallback_unit(options.fallback_format),
+        };
+        let mut prev_length = None;
+        let mut prev_symbol = None;
         for (length, symbol) in zone_symbols {
+            if prev_length == None && prev_symbol == None {
+                prev_length = Some(length);
+                prev_symbol = Some(symbol);
+            } else if prev_length != Some(length) && prev_symbol != Some(symbol) {
+                return Err(DateTimeFormatError::UnsupportedOptions);
```

Comment:
Suggestion: This should be a more specific error, and there should be a comment explaining what it means here.